meteor/sofie-server/data-access/repositories/test/mongo-segment-repository.spec.ts

```diff
@@ -159,9 +160,176 @@ describe(`${MongoSegmentRepository.name}`, () => {
+			})
+			await testee.save(inactiveSegment)
+
+			const result: MongoSegment = (await db
+				.collection(COLLECTION_NAME)
+				.findOne({ _id: id })) as unknown as MongoSegment
+
+			expect(result.isOnAir).toBeFalsy()
+		})
+
+		it('does not have segment as next but saves the segment as next', async () => {
+			const id: string = 'randomId'
+			const nonQueuedSegment: MongoSegment = createMongoSegment({ _id: id, isNext: false })
+			const nextSegment: Segment = createSegment({ id: id, isNext: true })
+
+			await testDatabase.populateDatabaseWithSegments([nonQueuedSegment])
+			const mongoConverter: MongoEntityConverter = mock(MongoEntityConverter)
+			const mongoDb: MongoDatabase = mock(MongoDatabase)
+			const db: Db = testDatabase.getDatabase()
+			const collection = db.collection(COLLECTION_NAME)
+
+			when(mongoDb.getCollection(anything())).thenReturn(collection)
+			when(mongoConverter.convertToMongoSegment(anything())).thenReturn({
+				_id: nextSegment.id,
+				_rank: nextSegment.rank,
+				externalId: '',
+				isHidden: false,
+				isNext: nextSegment.isNext(),
+				isOnAir: nextSegment.isOnAir(),
+				name: nextSegment.name,
+				rundownId: nextSegment.rundownId,
+			})
+
+			const testee: SegmentRepository = createTestee({
+				mongoDb: mongoDb,
+				mongoConverter: mongoConverter,
+			})
+			await testee.save(nextSegment)
+
+			const result: MongoSegment = (await db
+				.collection(COLLECTION_NAME)
+				.findOne({ _id: id })) as unknown as MongoSegment
+
+			expect(result.isNext).toBeTruthy()
+		})
+
+		it('has segment as next and saves the segment as not next', async () => {
+			const id: string = 'randomId'
+			const nextSegment: MongoSegment = createMongoSegment({ _id: id, isNext: true })
+			const nonQueuedSegment: Segment = createSegment({ id: id, isNext: false })
+
+			await testDatabase.populateDatabaseWithSegments([nextSegment])
+			const mongoConverter: MongoEntityConverter = mock(MongoEntityConverter)
+			const mongoDb: MongoDatabase = mock(MongoDatabase)
+			const db: Db = testDatabase.getDatabase()
+			const collection = db.collection(COLLECTION_NAME)
+
+			when(mongoDb.getCollection(anything())).thenReturn(collection)
+			when(mongoConverter.convertToMongoSegment(anything())).thenReturn({
+				_id: nonQueuedSegment.id,
+				_rank: nonQueuedSegment.rank,
+				externalId: '',
+				isHidden: false,
+				isNext: nonQueuedSegment.isNext(),
+				isOnAir: nonQueuedSegment.isOnAir(),
+				name: nonQueuedSegment.name,
+				rundownId: nonQueuedSegment.rundownId,
+			})
+
+			const testee: SegmentRepository = createTestee({
+				mongoDb: mongoDb,
+				mongoConverter: mongoConverter,
+			})
+			await testee.save(nonQueuedSegment)
+
+			const result: MongoSegment = (await db
+				.collection(COLLECTION_NAME)
+				.findOne({ _id: id })) as unknown as MongoSegment
+
+			expect(result.isNext).toBeFalsy()
+		})
+	})
+
+	// TODO: Extract to Helper Class in Model layer
+	function createSegment(params: {
```

Comment:
Should just be replaced by `EnityMockFactory`

Reply:
Replaced!